src/shared/organisms/DataExplorerGraphFlowNavigationStack/NavigationStack.spec.tsx

```diff
@@ -0,0 +1,260 @@
+        types: [],
+        resource: ['bbp', 'atlas', 'https://bbp.neuroshapes.org', 1],
+      })
+    );
+    rerender(app);
+    store.dispatch(
+      AddNewNodeDataExplorerGraphFlow({
+        isDownloadable: false,
+        _self:
+          'https://bbp.epfl.ch/nexus/v1/resources/neurosciencegraph/datamodels/_/https:%2F%2Fneuroshapes.org',
+        title: 'neuroshapes.org',
+        types: [],
+        resource: [
+          'neurosciencegraph',
+          'datamodels',
+          'https://neuroshapes.org',
+          161,
+        ],
+      })
+    );
+    render(app);
+    store.dispatch(AddNewNodeDataExplorerGraphFlow(fourthItemInStack));
+    render(app);
+    const navigationItemsAfterMultipleNavigation = container.querySelectorAll(
+      '.navigation-stack-item:not(.no-more)'
+    );
+    expect(navigationItemsAfterMultipleNavigation.length).toBe(4);
+    const state = store.getState();
+    expect(state.dataExplorer.links.length).toBe(4);
+  });
+  it('should render the NavigationStackShrinkedItem when it passed MAX_NAVIGATION_ITEMS_IN_STACK', () => {
+    store.dispatch(
+      AddNewNodeDataExplorerGraphFlow({
+        isDownloadable: false,
+        _self:
+          'https://bbp.epfl.ch/nexus/v1/resources/neurosciencegraph/datamodels/datashapes:ontologyentity/https:%2F%2Fbbp.epfl.ch%2Fontologies%2Fcore%2Fefeatures%2FNeuroElectroNeuronElectrophysiologicalFeature',
+        title: 'NeuroElectro Neuron Electrophysiological Feature',
+        types: ['Class'],
+        resource: [
+          'neurosciencegraph',
+          'datamodels',
+          'https://bbp.epfl.ch/ontologies/core/efeatures/NeuroElectroNeuronElectrophysiologicalFeature',
+          29,
+        ],
+      })
+    );
+    rerender(app);
+    store.dispatch(
+      AddNewNodeDataExplorerGraphFlow({
+        isDownloadable: false,
+        _self:
+          'https://bbp.epfl.ch/nexus/v1/resources/bbp/atlas/datashapes:atlasrelease/4906ab85-694f-469d-962f-c0174e901885',
+        title: 'Blue Brain Atlas',
+        types: ['AtlasRelease', 'BrainAtlasRelease'],
+        resource: [
+          'bbp',
+          'atlas',
+          'https://bbp.epfl.ch/neurosciencegraph/data/4906ab85-694f-469d-962f-c0174e901885',
+          3,
+        ],
+      })
+    );
+    rerender(app);
+    const navigationStackShrinkedItem = container.querySelectorAll(
+      '.navigation-stack-item.more'
+    );
+    expect(navigationStackShrinkedItem.length).toBe(1);
+    expect(navigationStackShrinkedItem).not.toBeNull();
+    const navigationHiddenItems = container.querySelectorAll(
+      '.navigation-stack-item:not(.no-more):not(.more)[hidden]'
+    );
+    expect(navigationHiddenItems.length).toBe(4);
+  });
+  it('should show the collapse button when the NavigationStackShrinkedItem is clicked', () => {
+    const navigationStackShrinkedItem = container.querySelector(
+      '.navigation-stack-item.more'
+    );
+    if (navigationStackShrinkedItem) {
```

Comment:
This looks wrong. Why would `navigationStackShrinkedItem` be falsy? If it is, this test is not really doing anything. We should set this test up in a way that `navigationStackShrinkedItem` is always in the DOM and then continue with the assertions.

Reply:
why it is wrong, we can test the `navigationStackShrinkedItem` if it's in the document but it will still a possibility to be null, due the response from `querySelector ` and for the test to complete we should have it in the document to continue